### ISSUE TITLE
Error in netcam_setup_ftp

### ISSUE THREAD

**Issue by bdn02** [Mr-Dave/motion#59]
I've found an error in netcam_setup_ftp method (netcam.c) in this piece of code:
```c
if (ftp_connect(netcam) < 0) {
ftp_free_context(netcam->ftp);
return -1;
}
```
The call to the ftp_free_context throw an "double free or corruption".
Thanks

**Comment by tosiara** [comment]
I never used FTP before, but I can confirm this is reproducible:
```
[0] [NTC] [ALL] conf_load: Processing thread 0 - config file motion.server.conf
[0] [NTC] [ALL] motion_startup: Motion Unofficial-Git-003a666 Started
[0] [NTC] [ALL] motion_startup: Logging to syslog
[0] [NTC] [ALL] motion_startup: Using log type (ALL) log level (NTC)
[0] [NTC] [ENC] ffmpeg_init: ffmpeg LIBAVCODEC_BUILD 3670372 LIBAVFORMAT_BUILD 3671141
[0] [NTC] [ALL] main: Thread 1 is from motion.server.conf
[0] [NTC] [ALL] main: Thread 1 is device: ftp://server/ input -1
[0] [NTC] [ALL] main: Stream port 8081
[New Thread 0x7ffff5a4b700 (LWP 2825)]
[1] [NTC] [ALL] motion_init: Thread 1 started , motion detection Enabled
[1] [ALR] [NET] netcam_start: Network Camera thread starting... for url (ftp://server/)
[New Thread 0x7ffff524a700 (LWP 2826)]
[0] [NTC] [ALL] main: Waiting for threads to finish, pid: 2821
[0] [NTC] [STR] httpd_run: motion-httpd testing : IPV4 addr: 127.0.0.1 port: 8080
[0] [NTC] [STR] httpd_run: motion-httpd Bound : IPV4 addr: 127.0.0.1 port: 8080
[0] [NTC] [STR] httpd_run: motion-httpd/Unofficial-Git-003a666 running, accepting connections
[0] [NTC] [STR] httpd_run: motion-httpd: waiting for data on 127.0.0.1 port TCP 8080
[1] [ERR] [NET] netcam_read_ftp_jpeg: ftp_get_socket failed
[1] [CRT] [NET] netcam_start: Failed trying to read first image - retval:-1

Program received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7ffff5a4b700 (LWP 2825)]
0x000000000047096e in netcam_start (cnt=cnt@entry=0x192a030) at netcam.c:2844
2844            netcam->rtsp->status = RTSP_NOTCONNECTED;
(gdb) backtrace
#0  0x000000000047096e in netcam_start (cnt=cnt@entry=0x192a030) at netcam.c:2844
#1  0x000000000046c924 in vid_start (cnt=cnt@entry=0x192a030) at video_common.c:834
#2  0x0000000000461f4f in motion_init (cnt=<optimized out>) at motion.c:699
#3  motion_loop (arg=0x192a030) at motion.c:1093
#4  0x00007ffff78c8e0f in start_thread () from /lib64/libpthread.so.0
#5  0x00007ffff637a0dd in clone () from /lib64/libc.so.6
```

**Comment by Mr-Dave** [comment]
Commit d71cebf8e84bd72585ab946a68649e7792bbd0c2 should resolve this and a couple of other issues with the ftp option.

**Comment by bdn02** [comment]
Ok, the problem is fixed, but i've found another problem.
If at motion startup the jpg file is not present i receive a segmentation fault:

[1] [ALR] [NET] netcam_start: Network Camera thread starting... for url (ftp://192.168.1.3/image1.jpg)
[1] [DBG] [NET] netcam_url_parse: Entry netcam_url_parse data ftp://192.168.1.3/image1.jpg
[1] [DBG] [NET] netcam_url_parse: Parse case 0 data ftp://192.168.1.3/image1.jpg
[1] [DBG] [NET] netcam_url_parse: Parse case 1 data ftp
[1] [DBG] [NET] netcam_url_parse: Parse case 6 data 192.168.1.3
[1] [DBG] [NET] netcam_url_parse: Parse case 9 data /image1.jpg
[1] [INF] [NET] netcam_start: Netcam_http parameter 'off' converts to flags: HTTP/1.0: 1 HTTP/1.1: 0 Keep-Alive OFF.
[1] [INF] [NET] netcam_start: now calling netcam_setup_ftp
[1] [ERR] [NET] netcam_read_ftp_jpeg: ftp_get_socket failed
[1] [CRT] [NET] netcam_start: Failed trying to read first image - retval:-1
Errore di segmentazione (core dump creato)

Thanks




----Messaggio originale----

Da: notifications@github.com

Data: 21/03/2015 14.29

A: "Mr-Dave/motion"<motion@noreply.github.com>

Cc: "bdn02"<bdn02@libero.it>

Ogg: Re: [motion] Error in netcam_setup_ftp (#59)




Commit d71cebf should resolve this and a couple of other issues with the ftp option.

**Comment by Mr-Dave** [comment]
I got this error during testing and it appears that you might not have retrieved the latest version of all the files from the unstable branch.  I thought that this error was fixed by the IF condition that was added to line 2852 of netcam.c  Also, the netcam_ftp.c doesn't appear to be updated from your log above since it should now be reporting debug messages from the ftp server.

**Comment by bdn02** [comment]
You are right, i've retrieved the latest unstable branch and now work fine.
Thank you





----Messaggio originale----

Da: notifications@github.com

Data: 24/03/2015 0.31

A: "Mr-Dave/motion"<motion@noreply.github.com>

Cc: "bdn02"<bdn02@libero.it>

Ogg: Re: [motion] Error in netcam_setup_ftp (#59)




I got this error during testing and it appears that you might not have retrieved the latest version of all the files from the unstable branch.  I thought that this error was fixed by the IF condition that was added to line 2852 of netcam.c  Also, the netcam_ftp.c doesn't appear to be updated from your log above since it should now be reporting debug messages from the ftp server.
